src/label_view.js

```diff
@@ -1,171 +1,184 @@
+  }
 
-function previewClicked(event) {
-  const id = $(event.target)
-    .parents("div")
-    .attr("id");
+  setPreviewed(id) {
+    this.port.postMessage({ setTrue: id });
+  }
 
-  setPreviewed(id);
-}
+  boxClicked(event) {
+    const id = $(event.target)
+      .parents("div")
+      .attr("id");
 
-function fillFrame(event) {
-  $(".preview-frame").html(""); // clear all iframes
-
-  let $preview = $(event.target)
-    .parents(".music-grid-item")
-    .find(".preview-frame");
-  const idAndType = $preview.attr("id");
-  const id = idAndType.split("-")[1];
-  const idType = idAndType.split("-")[0];
-
-  // determine if preview window needs to be open
-  if (previewOpen == true && previewId == id) {
-    previewOpen = false;
-  } else {
-    previewId = id;
-    previewOpen = true;
+    this.port.postMessage({ toggle: id });
   }
 
-  if (previewOpen) {
-    let $checkbox = $(event.target)
-      .parents(`[id='${id}']`)
-      .find(".historybox");
-
-    let url = `https://bandcamp.com/EmbeddedPlayer/${idType}=${id}`;
-    url +=
-      '/size=large/bgcol=ffffff/linkcol=0687f5/tracklist=true/artwork=none/transparent=true/"';
+  previewClicked(event) {
+    const id = $(event.target)
+      .parents("div")
+      .attr("id");
 
-    const iframe_style =
-      "margin: 6px 0px 0px 0px; border: 0; width: 150%; height: 300px; position:relative; z-index:1;";
-    const iframe_val = `<iframe style="${iframe_style}" src="${url}" seamless></iframe>`;
-    $preview.html(iframe_val);
+    this.setPreviewed(id);
   }
-}
 
-function generatePreview(id, idType) {
-  let $button = $("<button>")
-    .attr("title", "load preview player")
-    .attr("type", "button")
-    .attr("class", "follow-unfollow open-iframe")
-    .attr("style", "width: 90%");
-
-  let $preview = $("<div>").html("Preview");
-  $button.append($preview);
-
-  let $checkbox = $("<button>")
-    .attr("title", "preview history (click to toggle)")
-    .attr(
-      "style",
-      "margin: 0px 0px 0px 5px; width: 28px; height: 28px; position: absolute;"
-    )
-    .attr("class", "follow-unfollow historybox");
-
-  $preview = $("<div>")
-    .attr("class", "preview-frame")
-    .attr("id", `${idType}-${id}`);
-
-  let $parent = $("<div>")
-    .attr("id", id)
-    .attr("class", "preview")
-    .append($button)
-    .append($checkbox)
-    .append($preview);
-
-  return $parent;
-}
+  fillFrame(event) {
+    $(".preview-frame").html(""); // clear all iframes
 
-$(document).ready(function() {
-  // iterate over page to get album IDs and append buttons with value
-  $("li.music-grid-item[data-item-id]").each(function(index, item) {
-    const idAndType = $(item)
-      .closest("li")
-      .attr("data-item-id");
+    let $preview = $(event.target)
+      .parents(".music-grid-item")
+      .find(".preview-frame");
+    const idAndType = $preview.attr("id");
+    // Todo: what happens if id doesn't match this format?
```

Comment:
If this event target doesn't match the expected `idAndType` format, this method fails when trying to `.split('-')`. Should this be caught and handled?

Reply:
not sure BUT this format is generated by us so it should always match the format.

Reply:
Cool - opting to leave as-is 👍